plugin-server/src/worker/ingestion/process-event.ts

```diff
@@ -92,11 +94,27 @@ export class EventsProcessor {
                         this.pluginsServer.statsd?.timing('kafka_queue.single_save.snapshot', singleSaveTimer, {
                             team_id: teamId.toString(),
                         })
-                        // No return value in case of snapshot events as we don't do action matching on them
                     } finally {
                         clearTimeout(timeout2)
                     }
                 }
+            } else if (data['event'] === '$performance_event') {
```

Comment:
Sending these events in via the main topic is not a good idea or it sharing the rest of the ingestion pipeline.

Similar to session recordings, having ingesting these events be batched/buffered/create persons/update properties/be processed by plugins touches a very complex part of an application you don't want to be touching. While adding this here speeds you up short-term you're creating problems for yourself and #team-pipeline as soon as it needs to get untangled.

Also the capture endpoint has custom partitioning logic that's not relevant for performance event.

This needs a rethink: I suggest removing it from this PR or putting it behind a flag completely, with the understanding that this will be reviewed by the proper team.